### PR TITLE
fix(Storage): AL (asset library) methodology deprecated since iOS 8 

### PR DESCRIPTION
**Pull request by russellwheatley**
### Description

This PR stems from the deprecation of the AL (asset library) URL methodology for iOS that started in iOS 8, and now appears to be defunct in iOS 13.

The Storage package relies on this [method](https://developer.apple.com/documentation/photokit/phasset/1624782-fetchassetswithalasseturls) to fetch a local asset using the AL methodology. This method now returns a `Nil` object that was the probable cause for app crashes in https://github.com/invertase/react-native-firebase/issues/2762 & https://github.com/invertase/react-native-firebase/issues/4032.

This PR ensures any `Nil` objects resulting from the defunct AL methodology ought to safely return an error message to JS side.

### Related issues
https://github.com/invertase/react-native-firebase/issues/2762
https://github.com/invertase/react-native-firebase/issues/4032

### Checklist

- I read the [Contributor Guide](../CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
  - [X] Yes
- My change supports the following platforms;
  - [ ] `Android`
  - [X] `iOS`
- My change includes tests;
  - [ ] `e2e` tests added or updated in `packages/\*\*/e2e`
  - [ ] `jest` tests added or updated in `packages/\*\*/__tests__`
- [ ] I have updated TypeScript types that are affected by my change.
- This is a breaking change;
  - [ ] Yes
  - [X] No

### Test Plan

---

Think `react-native-firebase` is great? Please consider supporting the project with any of the below:

- 👉 Star this repo on GitHub ⭐️
- 👉 Follow [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter

### CODE REVIEW

**Comment by mikehardy** [comment]
not reviewing (per request not to) but this also caused a build failure on one of the other os as well (catalyst I think?) because it's so dead-dead :-)

**Comment by russellwheatley** [comment]
@mikehardy Yeah, I took a look at that [issue](https://github.com/invertase/react-native-firebase/issues/2698) when making this PR, it partly affects the same area of code!

File: packages/app/ios/RNFBApp/RNFBUtilsModule.m [hunk 1]
```diff
@@ -57,16 +57,16 @@ + (PHAsset *)fetchAssetForPath:(NSString *)localFilePath {
   if ([localFilePath hasPrefix:@"assets-library://"] || [localFilePath hasPrefix:@"ph://"]) {
     if ([localFilePath hasPrefix:@"assets-library://"]) {
       NSURL *localFile = [[NSURL alloc] initWithString:localFilePath];
-#if TARGET_OS_MACCATALYST
```
Comment (mikehardy) [inline packages/app/ios/RNFBApp/RNFBUtilsModule.m:60]: @andymatuschak or @brunolemos - both of you are interested in this working on mac catalyst, can you confirm this change won't be a regression for you? We're trying to remove the AL APIs entirely and this is an intermediate step, but I'm not sure if the `@available` check is strong enough vs the `#if TARGET_nnn` test

**Comment by mikehardy** [comment]
I'm inclined to merge this but I asked for a targetted review from the 2 known-interested catalyst folks. If I don't hear back I'll merge but hopefully one or both gives a thumbs up :-)

File: packages/app/ios/RNFBApp/RNFBUtilsModule.m (hunk 1 above)
Comment (brunolemos) [reply packages/app/ios/RNFBApp/RNFBUtilsModule.m:60]: Hey Mike, if the catalyst build passes that's good enough for my case. I don't think I use this specific feature related to this code.

Comment (mikehardy) [reply packages/app/ios/RNFBApp/RNFBUtilsModule.m:60]: Okay - that's the hard part, I'm not sure it does - we don't have CI coverage of a catalyst build (yet) and none of us are currently targeting that platform personally, so it's untested as of yet. I have some action items related to that, but @brunolemos if you have any links to good docs on how to target react-native at catalyst that would help me cover it in CI to prove this works and keep it squared away for the future

Comment (andymatuschak) [reply packages/app/ios/RNFBApp/RNFBUtilsModule.m:60]: Thanks for checking. This'll be fine!

Comment (mikehardy) [reply packages/app/ios/RNFBApp/RNFBUtilsModule.m:60]: Okay great - thanks a ton for giving it a look @andymatuschak - I'll mark this for merge+release then, and of course if I/we are wrong about it I'll be listening and we'll get it fixed up so catalyst works. Cheers

**Pull request merged**